pkg/engine/virtualmachinescalesets.go

```diff
@@ -106,6 +106,9 @@ func CreateMasterVMSS(cs *api.ContainerService) VirtualMachineScaleSetARM {
 
 	vmProperties := &compute.VirtualMachineScaleSetProperties{}
 
+	if masterProfile.PlatformFaultDomainCount != nil {
+		vmProperties.PlatformFaultDomainCount = to.Int32Ptr(int32(*masterProfile.PlatformFaultDomainCount))
```

Comment:
What does dereferencing and then casting back to an int32 pointer actually do here (as opposed to just passing through `*masterProfile.PlatformFaultDomainCount`?

Reply:
oh... int on one side int32 on the other side :/
so it takes the *int dereferences it to cast it to int32 and takes the pointer of that... pretty ugly